merlin-server/src/test/java/gov/nasa/jpl/aerie/merlin/server/services/ConstraintsDSLCompilationServiceTests.java

```diff
@@ -277,6 +278,31 @@ export default() => {
     );
   }
 
+  @Test
+  void testClampBy() {
+    checkSuccessfulCompilation(
+        """
+            export default() => {
+              return Real.Resource("state of charge").rate().equal(Real.Value(4.0)).shiftBy(1000, -200)
```

Comment:
(not blocker) Fun fact, you can omit `Real.Value` in the right-hand-side of all operations in the API :) So `.equal(4)` will work too.

Reply:
Nice feature !